File: 35-search-insert-position/35-search-insert-position.c


int searchInsert(int* nums, int numsSize, int target){
    int res;
    int low = 0;
    int high = numsSize -1;
    res = binary_search(nums, numsSize, low, high, target);
    return res;
}

int binary_search(int *nums, int numsSize,int low, int high, int target){
    int mid;
    if(high < low){
        return low;
    }
    mid = (low+high)/2;
    if(nums[mid]>target){
        return binary_search(nums, numsSize, low, mid-1, target);
    }else if(nums[mid]<target){
        return binary_search(nums, numsSize, mid+1, high, target);
    }else{
        return mid;
    }
}

File: 35-search-insert-position/35-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        left, right = 0, len(nums)
        while left < right:
            mid = left + (right - left) //2
            
            if target <= nums[mid]:
                right = mid 
            else:
                left = mid +1
        return left